cardano-cli/src/Cardano/CLI/Run/Key.hs

```diff
@@ -232,6 +255,126 @@ readExtendedVerificationKeyFile evkfile = do
+
+        wrapException :: Either MnemonicToSigningKeyError a -> ExceptT KeyCmdError IO a
+        wrapException = except . first KeyCmdMnemonicError
+
+    mnemonicWords <- readMnemonic mnemonicSource
+
+    case extendedSigningKeyType of
+      Cmd.ExtendedSigningPaymentKey paymentKeyNo ->
+        writeKeyToFile
+          =<< wrapException
+            ( signingKeyFromMnemonicWithPaymentKeyIndex
+                AsPaymentExtendedKey
+                mnemonicWords
+                derivationAccountNo
+                paymentKeyNo
+            )
+      Cmd.ExtendedSigningStakeKey paymentKeyNo ->
+        writeKeyToFile
+          =<< wrapException
+            ( signingKeyFromMnemonicWithPaymentKeyIndex
+                AsStakeExtendedKey
+                mnemonicWords
+                derivationAccountNo
+                paymentKeyNo
+            )
+      Cmd.ExtendedSigningDRepKey ->
+        writeKeyToFile
+          =<< wrapException (signingKeyFromMnemonic AsDRepExtendedKey mnemonicWords derivationAccountNo)
+      Cmd.ExtendedSigningCCColdKey ->
+        writeKeyToFile
+          =<< wrapException
+            (signingKeyFromMnemonic AsCommitteeColdExtendedKey mnemonicWords derivationAccountNo)
+      Cmd.ExtendedSigningCCHotKey ->
+        writeKeyToFile
+          =<< wrapException
+            (signingKeyFromMnemonic AsCommitteeHotExtendedKey mnemonicWords derivationAccountNo)
+   where
+    writeSigningKeyFile
+      :: (HasTextEnvelope (SigningKey a), SerialiseAsBech32 (SigningKey a))
+      => KeyOutputFormat -> SigningKeyFile Out -> SigningKey a -> ExceptT KeyCmdError IO ()
+    writeSigningKeyFile fmt sKeyPath skey =
+      firstExceptT KeyCmdWriteFileError $
+        case fmt of
+          KeyOutputFormatTextEnvelope ->
+            newExceptT $
+              writeLazyByteStringFile sKeyPath $
+                textEnvelopeToJSON Nothing skey
+          KeyOutputFormatBech32 ->
+            newExceptT $
+              writeTextFile sKeyPath $
+                serialiseToBech32 skey
+
+    readMnemonic :: Cmd.MnemonicSource -> ExceptT KeyCmdError IO [Text]
+    readMnemonic (Cmd.MnemonicFromFile filePath) = do
+      fileText <- firstExceptT KeyCmdReadMnemonicFileError $ except =<< readTextFile filePath
+      return $ map T.pack $ words $ T.unpack fileText
+    readMnemonic Cmd.MnemonicFromInteractivePrompt =
+      liftIO $ do
+        putStrLn ""
+        putStrLn "Please enter your mnemonic sentence."
+        putStrLn ""
+        putStrLn " - It should consist of either: 12, 15, 18, 21, or 24 words."
+        putStrLn " - To terminate, press enter on an empty line."
+        putStrLn " - To abort you can press CTRL+C."
+        putStrLn ""
+        putStrLn "(If your terminal supports it, you can use the TAB key for word completion.)"
+        putStrLn ""
+        runInputTBehaviorWithPrefs defaultBehavior defaultPrefs settings (inputT ("", "") [])
```

Comment:
Looks cool. Can we support non-interactive senario with reading of mnemonic from stdin here? I imagine the following example, when someone stores their mnemonic in bitwarden, and tries to generate key using the following snippet:
```bash
bw list items --search 'cardano super secret mnemonic' \
  | jq -r '.[0].login.password' \
  | cardano-cli latest key key-from-mnemonic ... --signing-key-file key.json
```

@CarlosLopezDeLara thoughts? do you think this could be useful?

Reply:
This would actually work with the interactive one as is, I made it with that in mind, and I tested it feeding the output file of the generate-mnemonic command. But it may be a good idea to make one that is explicitly for that, or adding a comment, because the user may not know.

Reply:
Added a comment in: https://github.com/IntersectMBO/cardano-cli/pull/975/commits/6a55eca735a82c6f5815dc1c1ca54daf033cf56f